app/workers/delete_service_hierarchy_worker.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+class DeleteServiceHierarchyWorker < DeleteObjectHierarchyWorker
+  def perform(*)
+    purge_backend_apis unless service.account.provider_can_use?(:api_as_product)
```

Comment:
Add a TODO here to remove this line when api_as_product is shipped
Otherwise we might just remove the conditional thus removing all shared BackendApis